crates/sql_provider_datafusion/src/lib.rs

```diff
@@ -53,9 +61,18 @@ impl<T, P> SqlTable<T, P> {
             .connect()
             .await
             .context(UnableToGetConnectionFromPoolSnafu)?;
-        let schema = conn
-            .get_schema(&table_reference)
-            .context(UnableToQueryDbConnectionSnafu)?;
+
+        let schema = if let Some(conn) = conn.as_sync() {
+            conn.get_schema(&table_reference)
+                .context(UnableToQueryDbConnectionSnafu)?
+        } else if let Some(conn) = conn.as_async() {
+            conn.get_schema(&table_reference)
+                .await
+                .context(UnableToQueryDbConnectionSnafu)?
+        } else {
+            return Err(Error::UnableToDowncastConnection);
+        };
```

Comment:
Instead of requiring the callers to have to do this - why don't we add a default function on the DbConnection trait that basically does this? Then at least from the caller's perspective, it always looks async.